app/models/user.rb

```diff
@@ -0,0 +1,38 @@
+require 'bcrypt'
+
+class User < ActiveRecord::Base
+  include BCrypt
+  
+  has_many :cars
+
+  validates :first_name, presence: true
+  validates :last_name, presence: true
+  validates :username, presence: true, uniqueness: true
+  validates :email, presence: true, uniqueness: true,
+    format: { with: /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i,
+    message: "The email is not properly formatted (i.e. joe.someguy@example.com)" }
+  validates :password_hash, presence: true
+
+  def add_car(car)
+    return "Your garage is full! You cannot add any more cars." if self.cars.length >= 5
+    self.cars << car
+  end
+
+  def password
+    @password ||= Password.new(password_hash)
+  end
+
+  def password=(new_password)
+    @password = Password.create(new_password)
+    self.password_hash = @password
+  end
+
+  def self.login(email, password)
```

Comment:
Authenticate would be a better name for this method. Calling it login implies that the session will be set by it, but that's not happening here. .authenticate makes a little more sense because you are verifying that the user is "authentic" or registered in your database.